### ISSUE TITLE
Расчет больничного

### ISSUE THREAD

**Issue by Linda4ka** [Contabilizare/c5#535]
Добрый день! Существует следующая ситуация:У сотрудника был период болезни с 05.01.22-15.01.22. Больничный лист он предоставил в июне ( это позволяет законодательство. Принести больничный лист можно в течении года после заболевания). Сумма данного больничного и период 05.01-15.01 должны попасть в IPC 21 в июне.Проработала данный случай в типовой и ничего. Больничный не рассчитался, сумма не попала в Выплату, период тоже не отразился в июньском IPC.

**Comment by Nullarity** [comment]
Подскажите пожалуйста, я правитьно понимаю, что в январе этот сотрудник недополучил заработную плату, так как отсутствовал по невыясненной причине?

**Comment by Nullarity** [comment]
> Проработала данный случай в типовой и ничего. Больничный не рассчитался, сумма не попала в Выплату, период тоже не отразился в июньском IPC.

Уточните пожалуйста, как именно вы отрабатывали этот сценарий?

В июньском начислении, попробуйе выполнить заполнение с фильтром по этому сотруднику, указав период заполнения - январь, с выключенной галочкой очистки табличной части.

**Comment by Linda4ka** [comment]
> Подскажите пожалуйста, я правитьно понимаю, что в январе этот сотрудник недополучил заработную плату, так как отсутствовал по невыясненной причине?

Нет, в данном случае сотрудник получит деньги за фактически отработанное время, т.к. в табеле у него эти дни будет пропуск, но без кода. Когда он приносит больничный, на предприятии происходит его начисление и выплата в том месяце, когда этот больничный зарегистрирован.

**Comment by Linda4ka** [comment]
> > Проработала данный случай в типовой и ничего. Больничный не рассчитался, сумма не попала в Выплату, период тоже не отразился в июньском IPC.
> 
> Уточните пожалуйста, как именно вы отрабатывали этот сценарий?
> 
> В июньском начислении, попробуйте выполнить заполнение с фильтром по этому сотруднику, указав период заполнения - январь, с выключенной галочкой очистки табличной части

Хорошо, попробую.

**Comment by Linda4ka** [comment]
Сделала, но вот какая штука, кроме больничного за январь потянуло еще и оклад за январь
![image](https://user-images.githubusercontent.com/84187165/174953397-f88831ea-2546-46da-97d9-ee863c6bb9c5.png)
а это уже точно лишнее.

**Comment by Linda4ka** [comment]
При этом , когда заполняем IPC 21 за июнь, то туда попадают не только период больничного за январь, как должно быть, а еще и рабочий период с суммами оклада, чего точно не должно быть.
![image](https://user-images.githubusercontent.com/84187165/174954738-8bc1a354-e3ab-4f63-b5b5-cbd9bc7ab90f.png)

**Comment by Nullarity** [comment]
Вышлите мне пожалуйста эту базу, у меня не получается воспроизвести проблему.

**Comment by Nullarity** [comment]
> Подскажите пожалуйста, я правитьно понимаю, что в январе этот сотрудник недополучил заработную плату, так как отсутствовал по невыясненной причине?

> Нет, в данном случае сотрудник получит деньги за фактически отработанное время, т.к. в табеле у него эти дни будет пропуск, но без кода. Когда он приносит больничный, на предприятии происходит его начисление и выплата в том месяце, когда этот больничный зарегистрирован.

Я тут не понял. Допустим 31 января выплатили зарплату, больной сотрудник получит зп за минусом дней болезни. Верно?

**Comment by Linda4ka** [comment]
Да, за фактически отработанные дни.

**Comment by Nullarity** [comment]
В приложенной базе, в январе было введено отсутствие на работе с десятого по четырнадцатое, а больничный с 5 по 15, здесь точно нет ошибки ввода данных?
В любом случае, в программе есть что улучшать на этот счёт (есть отдельная на это задача), но и имеющимися средствами ситуацию можно обыграть. Тонкость состоит в том, что документы отсутствие на работе нельзя ввести за выходной день. Для того, чтобы в июньское начисление зарплаты, попал только больничный, вам нужно в начале корректно ввести отсутствие на работе с пятого по седьмое и второй строкой с десятого по четырнадцатое января, затем произвести начисление зарплаты за январь, убедитесь предварительно чтобы больничный сняли с проведения. После этого, в табличной части начислений январского документа начисления зарплаты, нужно скорректировать одну запись, а именно, поставить период расчёта не с 15 января, а с шестнадцатого, так как сотрудник болел до пятнадцатого. Проведите после этого документ, и произведите начисление июньского начисления по указанному выше рецепту, то есть путём заполнения без очистки табличной части данных по выбранному сотруднику за январь. В этом случае, в июньское начисление должно добавиться только одно начисление по больничным.

**Comment by Linda4ka** [comment]
Когда в январе я захотела сделать отсутствия на все дни больничного, программа не позволила провести документ сославшись что по определенным датам нет часов. Поэтому пришлось вписывать дни отсутствия по тем дням, по которым они были.

**Comment by Linda4ka** [comment]
Процесс крайне сложный и не удобный. Надеюсь,, что данную проблему можно разрешить, тем более что это начисление обычного больничного и в данном случае расчет ведется за 3 месяца (март, апрель, май), т.е обычная схема.
Попробую сделать как сказали и позже напишу что получилось.
